### PR TITLE
fix(dashboard_monitor): correctly merge old and new sample data

### PR DESCRIPTION
**Pull request by savonarola**
Fixes [EMQX-12481](https://emqx.atlassian.net/browse/EMQX-12481), #13143 

Release version: v/e5.7.1

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [na] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [x] Schema changes are backward compatible

[EMQX-12481]: https://emqx.atlassian.net/browse/EMQX-12481?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by zmstone** [comment]
duplicate?  https://github.com/emqx/emqx/pull/13149

File: apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl [hunk 1]
```diff
@@ -191,6 +191,37 @@ t_monitor_sampler_format(_Config) ->
...
+            lists:seq(0, 9)
+        )
+    ),
+
+    ok = meck:new(emqx, [passthrough, no_history]),
+    ok = meck:expect(emqx, running_nodes, fun() -> [node(), 'other@node'] end),
+    ok = meck:new(emqx_dashboard_proto_v1, [passthrough, no_history]),
+    ok = meck:expect(emqx_dashboard_proto_v1, do_sample, fun('other@node', _Time) ->
+        FakeOldData
+    end),
```
Comment (keynslug) [inline apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl:211]: Q: Won't this testcase risk becoming false positive once, for example, dashboard starts to use `emqx_dashboard_proto_v2`?

**Review by keynslug — DISMISSED** [review]

**Review by zhongwencool — APPROVED** [review]

File: apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl (hunk 1 above)
Comment (keynslug) [reply apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl]: I suspect the _history_ feature of `meck` could be helpful here?

Comment (savonarola) [reply apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl]: Yes, there is an updated variant where the call to stub is pinned.

**Review by keynslug — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2592)

**Pull request merged**